pkg/dyninst/irgen/type_catalog.go

```diff
@@ -394,6 +395,11 @@ structChildren:
 					"failed to get name for member %q: %w", name, err,
 				)
 			}
+			// Special field that we don't currently have use for,
+			// but would need special handling to deal with duplicates.
+			if strings.HasPrefix(name, "_") {
```

Comment:
this is sadly wrong: only exactly `_` is ignored. `_foo` is totally fine... Let's add that to the test and make sure we capture it.

https://go.dev/play/p/dN20_KSyn9f